src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -674,9 +676,10 @@ private void filterAndAddKeys(DatastoreMetricQuery query, ResultSet rs, List<Dat
         }
 
         final boolean isCriticalQuery = rowReadCount > 5000 || filteredRowKeys.size() > 100;
-        if (isCriticalQuery) {
-            query.setCriticalQueryUUID(UUID.randomUUID());
-            logCriticalQuery(query, filteredRowKeys, rowReadCount, false, readRowsLimit, index);
+        if (isCriticalQuery || randomToLogQueriesOrNot.nextInt(100) < 10) {
```

Comment:
What do you think about extracting magic number 10 to configuration (e.g. environment variable)? Also, I'd simplify `randomToLogQueriesOrNot` to just `random` since in future it might be reused for other use cases (in the end, it's just a `new Random()` without further configurations).

Reply:
Make sense, will do